libraries/apollo-gradle-plugin-external/src/main/kotlin/com/apollographql/apollo3/gradle/internal/DefaultApolloExtension.kt

```diff
@@ -580,7 +732,17 @@ abstract class DefaultApolloExtension(
       }
       task.packageNameGenerator = packageNameGenerator
       task.generateFilterNotNull.set(project.isKotlinMultiplatform)
-      task.alwaysGenerateTypesMatching.set(service.alwaysGenerateTypesMatching)
+      task.alwaysGenerateTypesMatching.set(project.provider {
+        val jsonFiles = if (service.usedCoordinates != null) {
+          listOf(service.usedCoordinates!!)
+        } else {
+          usedCoordinatesConsumerConfiguration.files
+        }
+        jsonFiles.map { it.toUsedCoordinates() }
+            .fold(service.alwaysGenerateTypesMatching.getOrElse(emptySet())) { acc, new ->
+              acc + new
+            }
```

Comment:
(Nit again) this is the same as `service.alwaysGenerateTypesMatching.getOrElse(emptySet()) + jsonFiles.map { it.toUsedCoordinates() }`?

Reply:
Same thing. I think that'd return a `Set<Any>` ? Because it would try to add a `Iterable<Set<String>>` to a `Set<String>`

Reply:
Makes sense!